From 08f171a3c8bae0441a9b10cbca184b98db642895 Mon Sep 17 00:00:00 2001
From: "sadiqX, choudhari" <sadiqx.choudhari@intel.com>
Date: Fri, 30 Aug 2013 17:32:45 +0530
Subject: [PATCH] [PORT FROM MAIN]Enable use of OMX wrapped MDP decoder

BZ: 134846

This patch help the selection of soft OMX library which contains the name
mp3dec_mdp for component OMX.Intel.mp3.decoder and aacdec_mdp for
component OMX.Intel.aac.decoder

Severity: medium

Change-Id: Ic1b3b6b820d5d5ae6b615f15fb0f7581aa1f9d71
Orig-Change-Id: I8bc89c459c8b0687bff9fb5be594d320da2ce16d
Category: feature differentiation
Domain: AUDIO_MEDIA-Audio Codecs
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: sadiqX, choudhari <sadiqx.choudhari@intel.com>
---
 media/libstagefright/omx/Android.mk        | 4 ++++
 media/libstagefright/omx/SoftOMXPlugin.cpp | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/media/libstagefright/omx/Android.mk b/media/libstagefright/omx/Android.mk
index cd912e7..f93a2a1 100644
--- a/media/libstagefright/omx/Android.mk
+++ b/media/libstagefright/omx/Android.mk
@@ -27,6 +27,10 @@ LOCAL_SHARED_LIBRARIES :=               \
         libstagefright_foundation       \
         libdl
 
+ifeq ($(USE_INTEL_MDP),true)
+    LOCAL_CPPFLAGS += -DUSE_INTEL_MDP
+endif
+
 ifeq ($(BOARD_HAS_MTK_HARDWARE),true)
     LOCAL_C_INCLUDES += \
         $(TOP)/hardware/mediatek/media/include
diff --git a/media/libstagefright/omx/SoftOMXPlugin.cpp b/media/libstagefright/omx/SoftOMXPlugin.cpp
index c1c9ef6..3218801 100644
--- a/media/libstagefright/omx/SoftOMXPlugin.cpp
+++ b/media/libstagefright/omx/SoftOMXPlugin.cpp
@@ -55,6 +55,10 @@ static const struct {
     { "OMX.google.vp9.decoder", "vpxdec", "video_decoder.vp9" },
     { "OMX.google.vp8.encoder", "vpxenc", "video_encoder.vp8" },
     { "OMX.google.raw.decoder", "rawdec", "audio_decoder.raw" },
+#ifdef USE_INTEL_MDP
+    { "OMX.Intel.mp3.decoder", "mp3dec_mdp", "audio_decoder.mp3" },
+    { "OMX.Intel.aac.decoder", "aacdec_mdp", "audio_decoder.aac" },
+#endif
     { "OMX.google.flac.encoder", "flacenc", "audio_encoder.flac" },
     { "OMX.google.gsm.decoder", "gsmdec", "audio_decoder.gsm" },
 };
-- 
2.5.0

